### DISCUSSION TITLE
ui.refreshable strange behavior with ui.page

### DISCUSSION THREAD

**Discussion by TeeFiX42** [zauberzeug/nicegui#3221] [Q&A] [answered]
### Question

minimal code:
```py
@snip_app.page("/jos")
def jos():
    @ui.refreshable
    def jos_code():
        with snip_app.row():
            with ui.card():
                label = snip_app.label("Hello World")
            snip_app.button(text="Click me", on_click=lambda: label.set_text(str(datetime.datetime.now())))
    jos_code()
    ui.button("refresh", on_click=jos_code.refresh)
```

The above code works independant: When I open multiple tabs or browsers: all individual tabs have a private page.
But when I move
```py
    @ui.refreshable
    def jos_code():
        with snip_app.row():
            with ui.card():
                label = snip_app.label("Hello World")
            snip_app.button(text="Click me", on_click=lambda: label.set_text(str(datetime.datetime.now())))
```
to another python file and I import the function jos_code, the different tabs are no longer private: When I click on the refresh button in a TAB, all other TABs are also refreshed.

Can someone explain this, pls?

**Comment by python-and-novella** [comment]
Importing causes the scope to become a single Python file, and components or variables with a scope of a single file have the same performance

**Comment by falkoschindler** [comment, marked as answer]
Hi @TeeFiX42,

A function decorated with `ui.refreshable` will display the same content where ever it is called. To use the same implementation for different content, you somehow need to do the decoration in the page context. Here is a very similar question https://github.com/zauberzeug/nicegui/discussions/3233 where I showed two examples how to achieve that: https://github.com/zauberzeug/nicegui/discussions/3233#discussioncomment-9786799. Does that help to solve your problem?

**Comment by TeeFiX42** [comment]
Tnx. Good example. This solves our issue.Regards
